qiskit/transpiler/target.py

```diff
@@ -460,9 +460,10 @@ def update_from_instruction_schedule_map(self, inst_map, inst_name_map=None, err
     @property
     def qargs(self):
         """The set of qargs in the target."""
-        if None in self._qarg_gate_map:
+        qargs = set(self._qarg_gate_map)
+        if len(qargs) == 1 and next(iter(qargs)) is None:
```

Comment:
tests are still passing without this part of the condition:
```suggestion
        if len(qargs) == 1:
```

Reply:
It's an important part of the condition though, because if you have a target with only 1 qubit or only 1 gate on 1 qubit without the second half this would trigger incorrectly. For example,

```
target = Target()
target.add_instruction(HGate(), {(0,): None})
target.qargs
```

should return `{(0,),}` and not `None`. I can add a dedicated test for it if you think that's best

Reply:
I think so, because the second part of the condition is not currently covered otherwise.

Reply:
Added in https://github.com/Qiskit/qiskit-terra/pull/8977/commits/4e5e08c612c8a74502375923412a843184c8de94